Tests/test_AlignIO_FastaIO.py

```diff
@@ -2,73 +2,255 @@
 # This code is part of the Biopython distribution and governed by its
 # license.  Please see the LICENSE file that should have been included
 # as part of this package.
-"""Simple 'print-and-compare' unit test for fasta-m10 parser.
+"""Unit tests for the Bio.AlignIO.FastaIO module."""
 
-Created to check for any regressions from my new implementation of the
-parser.
-"""
-from __future__ import print_function
+import unittest
+from Bio.AlignIO import FastaIO
 
-import os
-from Bio import AlignIO
 
-# test_files is a list of tuples containing:
-# - string:  file format
-# - integer: number of sequences per alignment
-# - integer: number of alignments
-# - string:  relative filename
-#
-# Most of the input files are also used by test_SeqIO.py,
-# and by other additional tests as noted below.
-test_files = [
-    ("fasta-m10", 2, 4, 'Fasta/output001.m10'),
-    ("fasta-m10", 2, 6, 'Fasta/output002.m10'),
-    ("fasta-m10", 2, 3, 'Fasta/output003.m10'),
-    ("fasta-m10", 2, 1, 'Fasta/output004.m10'),
-    ("fasta-m10", 2, 1, 'Fasta/output005.m10'),
-    ("fasta-m10", 2, 1, 'Fasta/output006.m10'),
-    ("fasta-m10", 2, 9, 'Fasta/output007.m10'),
-    ("fasta-m10", 2, 12, 'Fasta/output008.m10'),
-    ]
+class FastaIOTests(unittest.TestCase):
+    """Test FastaIO module."""
 
-# Main tests...
-for (t_format, t_per, t_count, t_filename) in test_files:
-    assert t_format == "fasta-m10" and t_per == 2
+    def test_output001(self):
+        """Check output001.m10 file."""
+        fasta_file = 'Fasta/output001.m10'
+        with open(fasta_file, "r") as handle:
+            alignments = list(FastaIO.FastaM10Iterator(handle))
+            self.assertEqual(len(alignments), 4)
+            self.assertEqual(len(alignments[0]), 2)
+            self.assertEqual(alignments[0].get_alignment_length(), 108)
+            self.assertEqual(str(alignments[0][0].seq) + " " + alignments[0][0].id + " " + str(alignments[0][0].annotations["original_length"]), "SGSNT-RRRAISRPVRLTAEED---QEIRKRAAECGKTVSGFLRAAALGKKVNSLTDDRVLKEVM-----RLGALQKKLFIDGKRVGDREYAEVLIAITEYHRALLSR gi|10955263|ref|NP_052604.1| 107")
```

Comment:
That line and the others like it is far too long. Also it would be much clearer as three lines, and you shouldn't need ``str(...)`` on the sequence object.

```python
self.assertEqual(alignments[0][0].seq, "SGSNT-RRRAISRPVRLTAEED---QEIRKRAAE"
                                       "CGKTVSGFLRAAALGKKVNSLTDDRVLKEVM---"
                                       "--RLGALQKKLFIDGKRVGDREYAEVLIAITEYHRA"
                                       "LLSR")
self.assertEqual(alignments[0][0].id, "gi|10955263|ref|NP_052604.1|")
self.assertEqual(str(alignments[0][0].annotations["original_length"]), "107")
```

Do you need ``str(...)`` for the original length? Does this work?

```python
self.assertEqual(alignments[0][0].annotations["original_length"], 107)
```

Also since this becomes very repetitive and boring, I personally wouldn't check the exact sequence of every pairwise alignment. Focus on the interesting ones (e.g. gaps in odd places).

Reply:
Repetition is not a problem since I generate the test methods via a script.
If I don't use the `str(...)` I get back a `Seq` object type, which is not comparable with a string.

I can remove the `str(...)` from the original length.

Reply:
This should work on a Biopython 1.65 or later

```python
from Bio.Seq import Seq
assert Seq("ACGT") == "ACGT"
assert "ACGT" == Seq("ACGT")
```